_drafts/schematic_review_checklist_for_firmware_engineer.md

```diff
@@ -0,0 +1,132 @@
+
+However, I'd still have preferred a true reset line.
+
+## Story 4: GPIO Maps
+This is less of a war story, and more of a shoutout to good electrical engineers. Before walking into the schematic review for one project I was working on, the EE on the project published a spreadsheet of all the GPIOs for the MCU. This included their main function after reset, intended function, and any additional notes. It made it very easy as a firmware engineer to scan the list of GPIOs and ensure they were all configured properly. I'd recommend this is something that all firmware engineers ask for when looking at a schematic.
+
+## Story 5: Too much or too little debug
+I worked on a project once with an electrical engineer that proposed we make it hard to attach a debugger to the board. He proposed that we force someone to solder wires/leads to debug pads in order to get a debugger to work. As a young firmware engineer, I was floored. The debugger was my lifeline, and especially on a Cortex-M4 could help troubleshoot so many things. Making it difficult for a firmware engineer to use their most powerful tool seemed ludicras to me.
+
+However, the electrical engineers point was that firmware engineers rely too much on debuggers to debug devices, and don't spend enough time building tools/systems to track down crashes/bugs in the field, where a debugger is likely not attached. This was a good point. While working on fitness trackers years ago, I'd seen a firmware update over UART protocol continuously fail because none of the firmware developers used it (because it was broken of course). This meant it was only ever used by our QA team, who would file bugs, but it was never prioritized until one day it became a necessary firmware update path, and required a lot of scripting on the host side to work around firmware bugs.
+
+So I think that debuggers are essential tools for firmware engineers, and we should not hamper their usage. However, it's important to commit to building tools and systems for debugging systems in the field that use only the tools that will be available in the field, as your end goal as a firmware developer is to be able to fix those issues, not just the ones you can see at your desk.
+
+## Conclusion
+
+This is the start of my checklist for schematic reviews. Feel free to take it/modify, and add to it. I've found it useful to keep stories with rules to help me remember why the rules exist in the first place. All rules are meant to be broken, and sometimes I end up waiving items from the checklist, or adding new items for specific projects. These rules have helped me catch a few hardware/software interactions before boards were produced, which helped us keep to our schedule and get devices shipped on time!
```

Comment:
If you are interested, you could ask readers to comment what their schematic review horror stories are and what they now always check for

Reply:
Done! Thank you!